cppcache/src/ClientMetadataService.cpp

```diff
@@ -461,7 +468,13 @@ void ClientMetadataService::markPrimaryBucketForTimeout(
     std::shared_ptr<BucketServerLocation>& serverLocation, int8_t& version) {
   if (m_bucketWaitTimeout == std::chrono::milliseconds::zero()) return;
 
-  WriteGuard guard(m_PRbucketStatusLock);
+  //ReadGuard guard( m_PRbucketStatusLock );
+  std::unique_lock<std::timed_mutex> guard(m_timedBucketStatusLock, std::chrono::milliseconds(1000));
```

Comment:
Same comments above. If the same timeout is expected to occur in multiple locations then create a `constexp` for the timeout.